internal/qldb/account.go

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright 2023-present Coinbase Global, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package qldb
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"time"
+
+	"github.com/amzn/ion-go/ion"
+	"github.com/awslabs/amazon-qldb-driver-go/v3/qldbdriver"
+	"github.com/coinbase-samples/ib-ledger-go/internal/model"
+	"github.com/google/uuid"
+)
+
+var (
+    AccountNotFoundError = "account not found"
+)
+
+func CreateAccountAndInitializeBalance(
+	ctx context.Context,
+	userId, currency string,
+	initialBalance *big.Int) error {
+	_, err := Repo.Driver.Execute(ctx,
+		func(txn qldbdriver.Transaction) (interface{}, error) {
+			accountId := model.GenerateAccountId(userId, currency)
+			result, err := txn.Execute(`
+            SELECT * 
+            FROM Accounts 
+            WHERE id = ?`, accountId)
+			if err != nil {
+				return nil, err
+			}
+			if result.Next(txn) {
+				return nil, nil
+			}
+
+			initialDecimal := ion.NewDecimal(initialBalance, 0, false)
+			a := &model.QldbAccount{
+				Id:          model.GenerateAccountId(userId, currency),
+				Currency:    currency,
+				UserId:      userId,
+				Balance:     initialDecimal,
+				Hold:        ion.MustParseDecimal("0"),
+				Available:   initialDecimal,
+				UpdatedAt:   time.Now(),
+				AccountUUID: uuid.New().String(),
+			}
+
+			_, err = txn.Execute("INSERT INTO Accounts ?", a)
+
+			return nil, err
+		})
+
+	return err
+}
+
+func getAccountQldbTransaction(
+	txn qldbdriver.Transaction,
+	accountId string) (*model.QldbAccount, error) {
+	var account *model.QldbAccount
+	result, err := txn.Execute(`
+        SELECT * 
```

Comment:
Do we need multiline sql for something so simple?

Reply:
I was trying to make the line lengths shorter as they tended to go way over 80 which has been drilled into my brain as too long. I'll undo